salesforce_functions/data_api/_requests.py

```diff
@@ -173,9 +171,8 @@ def request_body(self) -> Json | None:
 
         for reference_id, sub_request in self._sub_requests.items():
             json_sub_request: dict[str, Any] = {
-                "url": sub_request.url(self._base_uri, self._api_version).removeprefix(
-                    self._base_uri
-                ),
+                # Sub-requests use relative URLs, hence the empty-string `org_domain_url`.
+                "url": sub_request.url("", self._api_version),
```

Comment:
Thought: could that be better modelled? Passing around an empty string feels incorrect to me.

Reply:
I see what you mean, however it feels like it would be adding quite a bit of boilerplate for minimal gain?

All five URL definitions would have to go from eg:

```python
    def url(self, org_domain_url: str, api_version: str) -> str:
        return f"{org_domain_url}/services/data/v{api_version}/sobjects/{self._object_type}/{self._record_id}"
```

To either something like this:

```python
    def url(self, org_domain_url: str | None, api_version: str) -> str:
        url_prefix = org_domain_url or ""
        return f"{url_prefix}/services/data/v{api_version}/sobjects/{self._record.type}"
```

Or:

```
    def url(self, org_domain_url: str | None, api_version: str) -> str:
        relative_api_path = f"/services/data/v{api_version}/sobjects/{self._record.type}"

        if org_domain_url:
            return f"{org_domain_url}{relative_api_path}"
        else:
            return relative_api_path
```

Perhaps a compromise would be to rename the `org_domain_url` parameter to `base_url` or `url_prefix` or similar? An empty string "prefix" seems a bit more natural?

Really though, I'm not sure of any decent way of writing these APIs given the current design choices (both in this package and upstream). Ideally the base URL and API version would always either be passed in the constructor of eg `QueryNextRecordsRestApiRequest` or else via `url()` alone - but neither can be used exclusively due to the way UnitOfWork has to work.

Reply:
Yeah, that seems like an unreasonably high cost to me too. Also, technically, an empty string is still a valid URL.